src/main/java/org/jenkinsci/plugins/workflow/graphanalysis/FlowScanningUtils.java

```diff
@@ -59,6 +62,45 @@ public boolean apply(FlowNode input) {
     // Default predicates, which may be used for common conditions
     public static final Predicate<FlowNode> MATCH_BLOCK_START = (Predicate)Predicates.instanceOf(BlockStartNode.class);
 
+    /** Sorts flownodes putting the one begun last (oldest startTime) at the end, with null times last
+     *  because likely they represent the newest nodes with a {@link TimingAction} not attached yet. */
+    public static final Comparator<FlowNode> TIME_ORDER_COMPARATOR = new Comparator<FlowNode>() {
+
+        /** Implements null checking because the use of this method will not easily permit FindBugs verification on NonNull*/
+        @Override
+        public int compare(@CheckForNull FlowNode first, @CheckForNull FlowNode second) {
+            if (first == null || second == null) {
+                return 0;  // Sorting by start time is 100% irrelevant
```

Comment:
Yeah, probably better to put nulls at the end since they reflect the newest nodes usually.